packages/block-editor/src/hooks/dimensions.js

```diff
@@ -39,15 +40,24 @@ import {
 	useIsPaddingDisabled,
 } from './padding';
 import useSetting from '../components/use-setting';
+import { store as blockEditorStore } from '../store';
 
 export const SPACING_SUPPORT_KEY = 'spacing';
 export const ALL_SIDES = [ 'top', 'right', 'bottom', 'left' ];
 export const AXIAL_SIDES = [ 'vertical', 'horizontal' ];
 
 function useVisualizerMouseOver() {
 	const [ isMouseOver, setIsMouseOver ] = useState( false );
-	const onMouseOver = () => setIsMouseOver( true );
-	const onMouseOut = () => setIsMouseOver( false );
+	const { startShowingVisualizer, stopShowingVisualizer } =
+		useDispatch( blockEditorStore );
+	const onMouseOver = () => {
+		startShowingVisualizer();
```

Comment:
This is an interesting problem to solve.

From the point of a public API, if a plugin calls `startShowingVisualizer` it has the effect of hiding the block toolbar, which I think would be unexpected. I see why it's like that though—`startTyping` also works the same way. It does leave open the possibility of triggering other effects when a user starts or stops typing.

I'd question if it's needed though, and I think code like this might cause the more confusion (e.g. I can imagine plugins calling `startTyping` to hide the toolbar when the user isn't actually typing, just to get the desired result). 

One option is that the actions/selectors/reducer can be more specific to what it does, e.g. `hideBlockToolbar` / `showBlockToolbar`.

That way there's the possibility of converging the `isTyping` code and the visualizer code to use the same system for hiding revealing the toolbar. Possibly deprecating `isTyping` along the way.

I think @ellatrix might be familiar with this code, so it'd be good to get a second opinion.

Reply:
yeh, as I was adding this I was thinking to myself "the naming isn't right", `hideBlockToolbar / showBlockToolbar` is much better from the external API point of view so have switched it to that.